### PR TITLE
#23537 select and assign  goal to experiment

### LINKED ISSUE #23537: [UI] Add Goal
### Is your feature request related to a problem? Please describe.
As a user can select a goal for the experiment

- REACH_PAGE
- BOUNCE_RATE

 `PATCH {{serverURL}}/api/v1/experiments/{{experimentId}}`
`{
    "goals": {
        "primary": {
            "name": "Reach thank-you page",
            "type": "REACH_PAGE",
            "conditions": [
                {
                    "parameter": "url",
                    "operator": "EQUALS",
                    "value": "thank-you"
                },
                {
                    "parameter": "referrer",
                    "operator": "EQUALS",
                    "value": "home"
                }
            ]
        }
    }
}`

### Describe the solution you'd like
Use a sidebar to show the 2 options

N/A

### Additional context
N/A

### Acceptance criteria
- Click on + set experiment Goal.
- Select the Goal Type

- Update the view

### PR DESCRIPTION
**Pull request by oidacra**
### Proposed Changes
* Enable button to add goal
* Show a sidebar to choose only one goal
* update the view
* show green tick in Goal title
* disabled link to add goal

### Checklist
- [X] Tests
- [X] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
N/A

### Screenshots
Original
Screenshot 2023-01-09 at 11 22 10 AM

Updated
Screenshot 2023-01-09 at 11 22 47 AM

Walkthrough

https://user-images.githubusercontent.com/1909643/211357881-64b7ced4-e39e-4a55-9dff-af73e31ce921.mov

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/store/dot-experiments-configuration-store.ts [hunk 1]
```diff
@@ -204,6 +226,44 @@ export class DotExperimentsConfigurationStore extends ComponentStore<DotExperime
...
+    // Goals
+    readonly setSelectedGoal = this.effect((selectedGoal$: Observable<{ goals: Goals }>) => {
+        return selectedGoal$.pipe(
+            tap(() =>
+                this.setSidebarStatus({
+                    status: Status.SAVING,
+                    experimentStep: ExperimentSteps.GOAL
+                })
+            ),
+            withLatestFrom(this.state$),
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/store/dot-experiments-configuration-store.ts:238]: Same as before, you should send this from the component.

**Review by fmontes — DISMISSED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/store/dot-experiments-configuration-store.ts (hunk 1 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/store/dot-experiments-configuration-store.ts]: Done.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/dot-experiments-configuration.component.ts [hunk 2]
```diff
@@ -100,8 +102,11 @@ export class DotExperimentsConfigurationComponent implements OnInit {
      * @returns void
```
Comment (alfredo-dotcms) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/dot-experiments-configuration.component.ts:102]: fix doc

Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/dot-experiments-configuration.component.ts:103]: Its already fixed

Comment (alfredo-dotcms) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/dot-experiments-configuration.component.ts:103]: no, you're missing `experimentId: string` in `deleteVariant` method

**Review by alfredo-dotcms — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Pull request merged**